cli/crates/typed-resolvers/src/codegen.rs

```diff
@@ -84,8 +84,21 @@ where
 fn render_graphql_type(r#type: &GraphqlType, schema: &AnalyzedSchema<'_>) -> String {
     let mut type_string = match &r#type.kind {
         TypeKind::BuiltinScalar(scalar) => match scalar {
-            BuiltinScalar::Int | BuiltinScalar::Float => "number",
-            BuiltinScalar::String | BuiltinScalar::Id => "string",
+            BuiltinScalar::Int
+            | BuiltinScalar::Float
+            | BuiltinScalar::Decimal
```

Comment:
I believe decimal is encoded as a string: https://github.com/grafbase/grafbase/blob/main/engine/crates/engine/src/registry/scalars/decimal.rs

Reply:
Indeed, fixing that.